java-generator/core/src/main/java/io/fabric8/java/generator/nodes/JObject.java

```diff
@@ -95,16 +106,31 @@ public JObject(
         if (!IGNORED_FIELDS.contains(field.getKey())) {
           String nextPrefix = (config.getPrefixStrategy() == Config.Prefix.ALWAYS) ? classPrefix : "";
           String nextSuffix = (config.getSuffixStrategy() == Config.Suffix.ALWAYS) ? classSuffix : "";
-
+          AbstractJSONSchema2Pojo fieldProp = AbstractJSONSchema2Pojo.fromJsonSchema(
+              field.getKey(),
+              field.getValue(),
+              nextPackagePath,
+              nextPrefix,
+              nextSuffix,
+              config);
+          // Let's have some additional criteria when generating default, i.e.: nodes with variable
```

Comment:
Yes, I decided to use the following:

`x-kubernetes-preserve-unknown-fields: true`

as an additional discriminant to include the default value into the representation. My intent was to skip objets that don't have a strict definition for default generation. Take the following case, an CRD `object` with no props and `x-kubernetes-preserve-unknown-fields: true`, see `readinessProbe` in akka-microservices CRD. It has a default to set `periodSeconds` but the generated object won't have that field and the code which would be generated to initialize the default would reference a non-existing field. And it won't compile.

I must admit aìthat a better criteria would rely on previous validation, but we don't have it. Maybe I should adjust the proposed changes in order to use a different condition, i.e. when `properties` is not defined) to better identify the case where no structure definition is provided. WDYT?

Reply:
Definitely this should be done differently, anyway. I am going to fix the criteria evaluation by removing all those RTTI based conditions and casts.

Reply:
Fixed, now as far as `object`s are concerned, the default is generated only for those with a defined set of `properties`, otherwise the generated code for the default to be set would not compile against the generated class code.